addons/pos_self_order/static/src/NavBar/NavBar.js

```diff
@@ -0,0 +1,21 @@
+/** @odoo-module */
+
+import { Component } from "@odoo/owl";
+import { useSelfOrder } from "@pos_self_order/SelfOrderService";
+export class NavBar extends Component {
+    static template = "NavBar";
+    setup() {
+        this.selfOrder = useSelfOrder();
+    }
+    functionToCallOnBackButton = () => {
+        if (this.props.previousPage) {
+            this.env.navigate(this.props.previousPage);
+            return;
+        }
+        if (this.props.executeFunction) {
```

Comment:
`executeFunction` is not passed to the component. You can remove this. If really needed, make sure to declare `static props` in the component so that we know which props it receives.

Reply:
And rename the props to `onClickBackButton`.